public/main.js

```diff
@@ -42,39 +65,89 @@ function reloadTodoList() {
     getTodoList(function(todos) {
         todoListPlaceholder.style.display = "none";
         var count = 0;
+        var completeList = [];
         todos.forEach(function(todo) {
+            if (!selector(todo, mode)) {
+                return;
+            }
             if (!todo.complete) {
                 count++;
             }
+            else {
+                completeList.push(todo);
+            }
             var listItem = document.createElement("li");
             listItem.textContent = todo.title;
             listItem.className = todo.complete ? "complete" : "incomplete";
 
-            var deleteButton = buttonFactory("delete", "delete-button", deleteEntry, todo);
+            var deleteButton = buttonFactory("delete", "button", deleteEntry, todo);
 
-            var completeButton = buttonFactory("complete", "complete-button", completeEntry, todo);
+            var completeButton = buttonFactory("complete", "button", completeEntry, todo);
```

Comment:
As buttons aren't using different classes any more initially, I think we should remove this parameter and hard code it in buttonFactory.